homeassistant/components/weather/__init__.py

```diff
@@ -221,6 +222,28 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         ],
         supports_response=SupportsResponse.ONLY,
     )
+    component.async_register_entity_service(
+        SERVICE_GET_FORECAST_VALUE,
+        {
+            vol.Required("attribute"): vol.In(
+                (
+                    "temperature",
+                    "templow",
+                    "precipitation",
+                    "precipitation_probability",
+                    "wind_speed",
+                    "humidty",
+                )
+            ),
+            vol.Optional("limit_hours"): vol.Range(1, 216),
+            vol.Optional("min_max", default="max"): vol.In(("min", "max")),
+        },
+        async_get_forecast_value_service,
+        required_features=[
+            WeatherEntityFeature.FORECAST_HOURLY,
```

Comment:
Why is this only supported for hourly forecast?

Reply:
Because of the hours based filter

Reply:
Right, but we know that the other forecasts are 12 and 24 hours respectively. Maybe the service handler should pick from available forecasts in the priority order hourly, twice daily, daily? @gjohansson-ST WDYT?

Reply:
I don't see a reason to limit this. I think we should support all forecast types but probably we leave it as an option in the schema but default to `hourly`

Reply:
Implemented FORECAST_TWICE_DAILY and FORECAST_DAILY